src/app/Agent/AgentLiveProbes.tsx

```diff
@@ -0,0 +1,266 @@
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableVariant,
+  ISortBy,
+  SortByDirection,
+  sortable,
+} from '@patternfly/react-table';
+import { useHistory } from 'react-router-dom';
+import { concatMap, filter, first } from 'rxjs/operators';
+import { LoadingView } from '@app/LoadingView/LoadingView';
+import { ErrorView, isAuthFail } from '@app/ErrorView/ErrorView';
+import { EventProbe } from '@app/Shared/Services/Api.service';
+
+export const AgentLiveProbes = () => {
+  const context = React.useContext(ServiceContext);
+  const history = useHistory();
+
+  const [templates, setTemplates] = React.useState([] as EventProbe[]);
+  const [filteredTemplates, setFilteredTemplates] = React.useState([] as EventProbe[]);
+  const [filterText, setFilterText] = React.useState('');
+  const [sortBy, setSortBy] = React.useState({} as ISortBy);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const addSubscription = useSubscriptions();
+
+  const tableColumns = [
+    { title: 'ID', transforms: [sortable] },
+    { title: 'Label', transforms: [sortable] },
+    { title: 'Class', transforms: [sortable] },
+    { title: 'Description', transforms: [sortable] },
+    { title: 'Method', transforms: [sortable] },
+  ];
+
+  React.useEffect(() => {
+    let filtered;
+    if (!filterText) {
+      filtered = templates;
+    } else {
+      const ft = filterText.trim().toLowerCase();
+      filtered = templates.filter(
+        (t: EventProbe) =>
+          t.name.toLowerCase().includes(ft) ||
+          t.description.toLowerCase().includes(ft) ||
+          t.clazz.toLowerCase().includes(ft) ||
+          t.methodDescriptor.toLowerCase().includes(ft) ||
+          t.methodName.toLowerCase().includes(ft)
+      );
+    }
+    const { index, direction } = sortBy;
+    if (typeof index === 'number') {
+      const keys = ['ID', 'Label', 'Description', 'Class', 'Method'];
+      const key = keys[index];
+      const sorted = filtered.sort((a, b) => (a[key] < b[key] ? -1 : a[key] > b[key] ? 1 : 0));
+      filtered = direction === SortByDirection.asc ? sorted : sorted.reverse();
+    }
+    setFilteredTemplates([...filtered]);
+  }, [filterText, templates, sortBy]);
+
+  const handleTemplates = React.useCallback(
+    (templates) => {
+      setTemplates(templates);
+      setErrorMessage('');
+      setIsLoading(false);
+    },
+    [setTemplates, setIsLoading, setErrorMessage]
+  );
+
+  const handleError = React.useCallback(
+    (error) => {
+      setErrorMessage(error.message);
+      setIsLoading(false);
+    },
+    [setIsLoading, setErrorMessage]
+  );
+
+  const refreshTemplates = React.useCallback(() => {
+    setIsLoading(true);
+    addSubscription(
+      context.target
+        .target()
+        .pipe(
+          filter((target) => target !== NO_TARGET),
+          first(),
+          concatMap((target) => context.api.getActiveProbes())
```

Comment:
We can remove call to `context.target.target` here as `getActiveProbes` already calls it internally.